--- src/System/System.cpp
+++ src/System/System.cpp
@@ -1,8 +1,11 @@
-#ifndef NDEBUG
+#ifdef _WIN32
+#  ifndef NDEBUG
 #define CRTDBG_MAP_ALLOC
 #include <stdlib.h>
 #include <crtdbg.h>
-#endif
+#  endif
+#  define UNICODE
+#endif

 #include <System/System.h>
 #include <System/Resources.h>
@@ -17,8 +20,6 @@
 #include <Editor/Editor.h>
 #include <Editor/Menubar.h>

-#define UNICODE
-
 namespace Vortex {
 namespace System {

--- src/System/Debug.cpp
+++ src/System/Debug.cpp
@@ -2,12 +2,14 @@

 #include <Core/WideString.h>

+#ifdef _WIN32
 #include <io.h>
 #include <fcntl.h>
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
 #include <stdio.h>
 #include <chrono>
-
-#define WIN32_LEAN_AND_MEAN
-#include <windows.h>
 #include <GL/gl.h>
 #undef ERROR
--- src/System/Mixer.cpp
+++ src/System/Mixer.cpp
@@ -2,14 +2,21 @@

 #include <malloc.h>

+#ifdef _WIN32
 #include "windows.h"
 #include "mmsystem.h"
+#else
+#include <pthread.h>
+#endif

 namespace Vortex {

 static const int WAVEOUT_CHANNELS = 2;
 static const int WAVEOUT_BLOCKS = 8;
 static const int WAVEOUT_BLOCK_FRAMES = 8192;
 static const int WAVEOUT_BLOCK_SIZE =
     sizeof(short) * WAVEOUT_CHANNELS * WAVEOUT_BLOCK_FRAMES;

+#ifdef _WIN32
 static void CALLBACK MixerCallback(HWAVEOUT hwo, UINT msg, DWORD_PTR, DWORD_PTR,
                                    DWORD_PTR);
--- src/System/Thread.cpp
+++ src/System/Thread.cpp
@@ -8,7 +8,9 @@

 #include <assert.h>

+#ifdef _WIN32
 #include <windows.h>
+#endif

 namespace Vortex {
