--- CMakeLists.txt
+++ CMakeLists.txt
@@ -10,19 +10,6 @@

 set(CMAKE_CONFIGURATION_TYPES Debug;Release;MinSizeRel)

-# clang-tidy might not be available on AppleClang systems by default
-find_program(CLANG_TIDY_EXE NAMES "clang-tidy")
-if(CLANG_TIDY_EXE)
-    if("${CMAKE_BUILD_TYPE}" STREQUAL "MinSizeRel")
-        set(tidy_flags "--fix-errors;--fix-notes")
-        set(format_flags "-i")
-    else()
-        set(tidy_flags "--warnings-as-errors=*")
-        set(format_flags "--dry-run")
-    endif()
-    set(CMAKE_CXX_CLANG_TIDY "clang-tidy;${tidy_flags};-extra-arg=/EHsc;-p=build")
-    message(STATUS "clang-tidy found: ${CLANG_TIDY_EXE}")
-else()
-    message(STATUS "clang-tidy not found, skipping static analysis")
-endif()
-
 include_directories("${PROJECT_SOURCE_DIR}/src")
